sklearn/cluster/hierarchical.py

```diff
@@ -712,7 +712,7 @@ class AgglomerativeClustering(BaseEstimator, ClusterMixin):
     n_leaves_ : int
         Number of leaves in the hierarchical tree.
 
-    n_components_ : int
+    n_connected_components_ : int
```

Comment:
You still need to keep `n_compenents_` in the `Attributes` section of the docstring.

See e.g. the deprecated attributes of `OneHotEncoder` in `preprocessing/_encoders.py`:

```
active_features_ : array
        Indices for active features, meaning values that actually occur
        in the training set. Only available when n_values is ``'auto'``.

        .. deprecated:: 0.20
            The ``active_features_`` attribute was deprecated in version
            0.20 and will be removed in 0.22.
```

Reply:
Thanks Nicolas, this was addressed in commit [#5626fbc](https://github.com/scikit-learn/scikit-learn/pull/13427/commits/5626fbc58b1ded04a77c11f7c7e53aa124e27fda)